### PR TITLE
Updating S3 multipart download

### LINKED ISSUE #52: Use EitherT with proper errors for download
https://github.com/ambiata/mismi/blob/c050946f194741c7f0a15707fcd13e59287fdd83/src/Mismi/S3/Commands.hs#L65

### PR DESCRIPTION
**Pull request by nhibberd**
Fixes #52 

On top of #241 and #242

### CODE REVIEW

File: mismi-s3/src/Mismi/S3/Commands.hs [hunk 1]
```diff
@@ -466,74 +476,61 @@ listRecursively a =
...
+    SyncInvariant a b ->
+      "Remove common prefix invariant: " <>
+      "[" <> addressToText b <> "] is not a common prefix of " <>
+      "[" <> addressToText a <> "]"
+    OutputExists a ->
+      "Can not copy to an address that already exists [" <> addressToText a <> "]"
+    SyncAws e ->
+      "AWS failure during 'sync': " <> renderError e
+
+worker :: Address -> Address -> SyncMode -> Env -> Address -> EitherT SyncWorkerError IO ()
```
Comment (markhibberd) [inline mismi-s3/src/Mismi/S3/Commands.hs]: :100:

File: mismi-s3/src/Mismi/S3/Commands.hs [hunk 2]
```diff
@@ -466,74 +476,61 @@ listRecursively a =
...
+worker input output mode env f = runAWST env SyncAws $ do
+  n <- maybe (left $ SyncInvariant input f) right $ removeCommonPrefix input f
+  let out = withKey (</> n) output
+      cp = lift $ copy f out
+  foldSyncMode
+    (ifM (lift $ exists out) (left $ OutputExists out) cp)
+    cp
```
Comment (markhibberd) [inline mismi-s3/src/Mismi/S3/Commands.hs]: Mmmm, this looks wrong, isn't this meant to be overwrite? Won't this check and fail?

Comment (nhibberd) [inline mismi-s3/src/Mismi/S3/Commands.hs]: Yes, this has been fixed in another PR (this needs to rebased)

Comment (nhibberd) [inline mismi-s3/src/Mismi/S3/Commands.hs]: NB - https://github.com/ambiata/mismi/pull/242/files#diff-1edb1b9e0d390245ba2b012dc71d0c42R521

**Comment by markhibberd** [comment]
:+1: Looks good once :green_apple: and few notes addressed.

**Pull request merged**